app/script/main/app.js

```diff
@@ -559,48 +558,19 @@ z.main.App = class App {
    * Check that this is the single instance tab of the app.
    * @returns {Promise} Resolves when page is the first tab
    */
-  _checkSingleInstanceOnInit() {
-    if (!z.util.Environment.electron) {
-      return this._setSingleInstanceCookie();
-    }
-
-    return Promise.resolve();
-  }
-
-  _checkSingleInstanceOnInterval() {
-    const singleInstanceCookie = Cookies.getJSON(App.CONFIG.TABS_CHECK.COOKIE_NAME);
-
-    const shouldBlockTab = !singleInstanceCookie || singleInstanceCookie.appInstanceId !== this.instanceId;
-    if (shouldBlockTab) {
-      return this._redirectToLogin(z.auth.SIGN_OUT_REASON.MULTIPLE_TABS);
+  _registerSingleInstance() {
+    if (this.singleInstanceHandler.registerInstance(this.instanceId)) {
+      this.singleInstanceHandler.addExtraInstanceStartedListener(this._onExtraInstanceStarted);
+      this._registerSingleInstanceCleaning();
+      return Promise.resolve();
     }
+    return Promise.reject(new z.auth.AuthError(z.auth.AuthError.TYPE.MULTIPLE_TABS));
   }
 
-  /**
-   * Set the cookie to verify we are running a single instace tab.
-   * @returns {undefined} No return value
-   */
-  _setSingleInstanceCookie() {
-    const shouldBlockTab = !!Cookies.get(App.CONFIG.TABS_CHECK.COOKIE_NAME);
-    if (shouldBlockTab) {
-      return Promise.reject(new z.auth.AuthError(z.auth.AuthError.TYPE.MULTIPLE_TABS));
-    }
-
-    const cookieData = {appInstanceId: this.instanceId};
-    Cookies.set(App.CONFIG.TABS_CHECK.COOKIE_NAME, cookieData);
-
-    window.setInterval(() => this._checkSingleInstanceOnInterval(), App.CONFIG.TABS_CHECK.INTERVAL);
-    this._registerSingleInstanceCookieDeletion();
-  }
-
-  _registerSingleInstanceCookieDeletion() {
+  _registerSingleInstanceCleaning(singleInstanceCheckIntervalId) {
     $(window).on('beforeunload', () => {
-      const singleInstanceCookie = Cookies.getJSON(App.CONFIG.TABS_CHECK.COOKIE_NAME);
-
-      const isOwnInstanceId = singleInstanceCookie && singleInstanceCookie.appInstanceId === this.instanceId;
-      if (isOwnInstanceId) {
-        Cookies.remove(App.CONFIG.TABS_CHECK.COOKIE_NAME);
-      }
+      this.singleInstanceHandler.removeExtraInstanceStartedListener(this._onExtraInstanceStarted);
+      this.singleInstanceHandler.deregisterInstance();
```

Comment:
We can also pass the listener(s) we intend to deregister with this. No need to an extra API call.